docs/spec/src/protocol/architecture/write-and-read-workflow.md

```diff
@@ -0,0 +1,57 @@
+
+4. **Disperser Constructs Blob Certificate.**
+   The disperser constructs a `BlobCertificate` consisting of the `BlobHeader` and a `RelayKey`, which can be used to identify the relay URI where the associated chunks are available.
+
+5. **Disperser Constructs Batch Header.**
+   The disperser constructs a `BatchHeader` consisting of a Merkelized collection of `BlobCertificate`s and a `ReferenceBlockNumber`, which anchors all blobs in the batch to a specific stake distribution on EigenLayer.
+
+6. **Disperser Sends Batch Header (W9).**
+   The disperser sends the `BatchHeader` to the validators using the `StoreChunks` API.
+
+7. **Validators Validate Batch Header.**
+   The validators validate the `PaymentMetadata` for each `BlobHeader` contained in the batch. If any blob contains improper payment information, the batch is rejected.
+
+8. **Validators Download and Validate Chunks (W10, W11).**
+   For properly authorized batches, validators reference the Chunk Assignment Logic together with the `QuorumNumbers` of each `BlobHeader` to determine which chunks they are responsible for hosting. Validators request all associated encoded chunks from the `GetChunks` interface of the appropriate relays and validate that each `Chunk` matches the corresponding blob's KZG commitment using the included opening proof. Validators also validate that each chunk has the correct length using the Chunk Assignment Logic. If any chunk is unavailable or cannot be validated, the batch is rejected.
+
+9. **Validators Sign Batch Header (W12).**
+   For batches that successfully complete validation, each validator signs the batch header using the BLS identity registered in the `EigenDAServiceManager` and returns the signature to the disperser.
+
+10. **Disperser Aggregates Signatures.**
+    The disperser aggregates the BLS signatures from the validators and returns a `Certificate` containing the `BatchHeader`, aggregate signature, and inclusion information used for verifying that a blob is part of the batch.
+
+
+### Read
+
+To read a blob, a client follows the sequence below. The labels in parentheses (e.g., R1, R2) correspond to the steps shown in the diagram above.
+
+1. **Read from Relay (R1).**
+   The client attempts to retrieve the blob from the `GetBlob` interface of the relay(s) identified in the `BlobHeader`. This is the primary and most efficient retrieval method, as the relay stores complete blobs.
+
+2. **Read from Validators (R2).**
+   If the blob is not available from the relay(s), the client falls back to retrieving individual chunks directly from the validators and reconstructing the blob. The client iterates through the `QuorumNumbers` in the `BlobHeader`. For each quorum (until successful recovery), the client requests chunks from each validator registered with that quorum, validates the integrity of the chunks, and then reconstructs the blob using the erasure coding scheme.
```

Comment:
We don't do a `for each quorum` when reading from the validators. Instead, the client reconstructs the blob assignments for all quorums, and downloads from validators in a random order until it has enough chunks.